libraries/browser-tracker-core/src/helpers/browser_props.ts

```diff
@@ -55,5 +58,18 @@ function detectDocumentSize() {
     bodyHeight = be ? Math.max(be.offsetHeight, be.scrollHeight) : 0;
   var w = Math.max(de.clientWidth, de.offsetWidth, de.scrollWidth);
   var h = Math.max(de.clientHeight, de.offsetHeight, de.scrollHeight, bodyHeight);
-  return isNaN(w) || isNaN(h) ? '' : w + 'x' + h;
+  return isNaN(w) || isNaN(h) ? '' : w + DIMENSION_SEPARATOR + h;
+}
+
+function detectScreenResolution() {
+  return screen.width + DIMENSION_SEPARATOR + screen.height;
+}
+
+export function floorDimensionFields(field?: string | null) {
```

Comment:
Nit: this seems a bit cumbersome to go split the string after previously joining it in the same code. Could we instead define a function that would round the values before joining them? It could be called from the `detectDocumentSize`, `detectViewport` and `detectScreenResolution` functions instead of concatenating the values directly. I mean something like:

```ts
function toDimension(x: number, y: number): string {
   return Math.floor(x) + 'x' + Math.floor(y);
}
```

Reply:
There were a couple of options on my mind, one being this one.

My thought process went like this:
- It seems 'wasted work' to join the string at some point and split it again.
- If this function is inside the other one's e.g. `detectDocumentSize`, `detectViewport` and `detectScreenResolution` , it will not need this _wasted_ join but it will not be clearly evident that you need to clean these types of values before sending them. By adding this function directly on the place that we send these properties makes it a bit more evident that if we are to add a new one, the same filter should apply.

Wdyt, is it worth it or more confusing ? 

I am all in for changing it as well

Reply:
Yeah I think it'd be nicer if `detectDocumentSize`, `detectViewport`, and `detectScreenResolution` all just return `[number, number] | null` and `toDimension()` just took that and returned `string | null`. Best of both worlds.